File: core/sources/list.ts
import { IndexedSourceBase, type Metadata } from "../types/source";

/**
 * Source containing a list of items.
 */
export class ListSource extends IndexedSourceBase {
  readonly kind = "list";

  /**
   * @param id Unique source ID
   * @param metadata Generated metadata for the source
   * @param content List items
   */
  constructor(
    id: string,
    metadata: Metadata | undefined,
    private readonly content: string[],
  ) {
    super(id, metadata);
  }

  get isIterable(): true {
    return true;
  }

  /** Returns list items. */
  getContent: () => string[] = () => this.content;

  buildIndexInput(): string {
    return this.content.join("\n");
  }

  buildDeepJudgeInput(): string {
    return this.content.slice(0, 10).join("\n");
  }
}

File: python/src/contextrie/sources/list.py
from __future__ import annotations

from ..types import IndexedSourceBase, Metadata


class ListSource(IndexedSourceBase):
    """Source containing a list of items."""

    kind = "list"

    def __init__(
        self,
        id: str,
        metadata: Metadata | None,
        content: list[str],
        path: str | None = None,
    ) -> None:
        super().__init__(id=id, metadata=metadata, path=path)
        self._content = content

    @property
    def is_iterable(self) -> bool:
        return True

    def get_content(self) -> list[str]:
        return self._content

    async def build_index_input(self) -> str:
        return "\n".join(self._content)

    async def build_deep_judge_input(self) -> str:
        return "\n".join(self._content[:10])
